### ISSUE TITLE
Captha refresh bug

### ISSUE THREAD

**Issue by ulrobix** [yiisoft/yii2#11181]
Same ticket https://github.com/yiisoft/yii2/issues/3601

After refresh captcha, the validator is always return an error.
Reproduced if using the route module 'main'.

1) I tired:
```
<?= $form->field($model, 'verifyCode')->widget(Captcha::className(), [
        'captchaAction' => 'activate-card/captcha'
    ])  ?>
```
rules:
`         ['verifyCode', 'captcha', 'captchaAction' => 'main/activate-card/captcha'],
` 
result:
js captha settings:
`{"refreshUrl":"/main/activate-card/captcha?refresh=1","hashKey":"yiiCaptcha/activate-card/captcha"});`
validator option hashKey:
`yiiCaptcha/main/activate-card/captcha`

 $('body').data(options.hashKey); in yii.validation.js not same with $('body').data(settings.hashKey, [data.hash1, data.hash2]); in yii.captha.js

2) I tried to specify the full path

```
<?= $form->field($model, 'verifyCode')->widget(Captcha::className(), [
        'captchaAction' => 'main/activate-card/captcha'
    ])  ?>

```

result:
Image not found. Url: /main/main/activate-card/captcha?v=56f327fe35936
js captha settings:
`{"refreshUrl":"/main/main/activate-card/captcha?refresh=1","hashKey":"yiiCaptcha/main/activate-card/captcha"}`

3) If set path with / still not working

```
<?= $form->field($model, 'verifyCode')->widget(Captcha::className(), [
        'captchaAction' => '/main/activate-card/captcha'
    ])  ?>

```
result:
{"refreshUrl":"/main/activate-card/captcha?refresh=1","hashKey":"yiiCaptcha//main/activate-card/captcha"}
Just get an error...

4). If you set captchaAction  'activate-card/captcha' in the model rules is getting the error 500.

In my case, it helped fix:
Class yii\captcha\Captcha
```
142:    protected function getClientOptions()
153: -            'hashKey' => "yiiCaptcha/{$route[0]}",
153: +             'hashKey' => "yiiCaptcha".Url::toRoute($this->captchaAction),
```

| Q                | A
| ---------------- | ---
| Yii version      |    2.0.7
| PHP version      |   5.5.9
| Operating system | Ubuntu 14.04

**Comment by SilverFire** [comment]
Fixed in #10910
Try to update to 2.0.8-dev

**Comment by ulrobix** [comment]
Thanks, ok
